singularity-eos/closure/mixed_cell_models.hpp

```diff
@@ -825,16 +931,30 @@ class PTESolverRhoT
     // Fill in the Jacobian
     for (std::size_t i = 0; i < neq * neq; ++i)
       jacobian[i] = 0.0;
+    // first row is energy residual eqn
+    jacobian[0] = dedT_sum;
+    std::size_t col = 1;
     for (std::size_t m = 0; m < nmat; ++m) {
-      jacobian[m] = 1.0;
-      jacobian[neq + m] = dedv[m];
+      if (m == ms) continue;
+      jacobian[col++] = dedv[m] - dedv[ms];
     }
-    jacobian[neq + nmat] = dedT_sum;
```

Comment:
Note that now $T$ is the 0th element of the state vector, rather than the last. I don't have a great reason for doing this, other than I felt the code was a little more legible, and I thought the Jacobian looked slightly nicer when I printed it out. I don't think it actually matters. :shrug: